callbacks/segmentation.py

```diff
@@ -1,50 +1,147 @@
+                "kwargs": {
+                    "job_type": "train",
+                    "experiment_id": "124",
+                    "dataset": "name_of_dataset",
+                    "params": '{"show_progress": 1}',
+                },
+            },
+        },
+    ],
+    "host_list": ["vaughan.als.lbl.gov"],
+    "dependencies": {"0": [], "1": [0]},
+    "requirements": {"num_processors": 2, "num_gpus": 0, "num_nodes": 1},
+}
 
 
-# NEXT STEPS:
-# - this function returns a job ID, which would be associated with the workflow run on vaughan
-# - then we need another callback to pick up this ID and start polling for successful output
 @callback(
-    Output("output-placeholder", "children"),
+    Output("output-details", "children"),
+    Output("submitted-job-id", "data"),
     Input("run-model", "n_clicks"),
     State("annotation-store", "data"),
     State("project-name-src", "value"),
 )
 def run_job(n_clicks, annotation_store, project_name):
-    # As a placeholder, pulling together the inputs we'd need if we were going to submit a job
+    """
+    This callback collects parameters from the UI and submits a job to the computing api.
+    If the app is run from "dev" mode, then only a placeholder job_uid will be created.
+    The job_uid is saved in a dcc.Store for reference by the check_job callback below.
+
+    # TODO: Appropriately paramaterize the DEMO_WORKFLOW json depending on user inputs
+    and relevant file paths
+    """
     if n_clicks:
+        if MODE == "dev":
+            job_uid = str(uuid.uuid4())
+            return (
+                dmc.Text(
+                    f"Workflow has been succesfully submitted with uid: {job_uid}",
+                    size="sm",
+                ),
+                job_uid,
+            )
+        else:
+            data_utils.save_annotations_data(annotation_store, project_name)
+            job_submitted = requests.post(
+                "http://job-service:8080/api/v0/workflows", json=DEMO_WORKFLOW
+            )
+            job_uid = job_submitted.json()
+            if job_submitted.status_code == 200:
+                return (
+                    dmc.Text(
+                        f"Workflow has been succesfully submitted with uid: {job_uid}",
+                        size="sm",
+                    ),
+                    job_uid,
+                )
+            else:
+                return (
+                    dmc.Text(
+                        f"Workflow presented error code: {job_submitted.status_code}",
+                        size="sm",
+                    ),
+                    job_uid,
+                )
+    return no_update, no_update
+
+
+@callback(
+    Output("output-details", "children", allow_duplicate=True),
+    Output("submitted-job-id", "data", allow_duplicate=True),
+    Input("submitted-job-id", "data"),
+    Input("model-check", "n_intervals"),
+    prevent_initial_call=True,
+)
+def check_job(job_id, n_intervals):
```

Comment:
This is redundant because the models will most likely take a long time to finish and the user won't wait for it.
It could be worth adding a notification that shows while they are away (or the first time they come back - would need to track in the DB if the notification has been seen) that would inform them that since their last visit, the ML job(s) has(have) finished and list them.

2 weeks when we talked about showing ML output, we agreed that it would go under `Data Selection`, and if the particular project has ML output, there would appear another dropdown where users would be able to select what they wish to view, definitely not a toggle since they are not limited to 1 ML output per project. See #62 I edited the Issue description with the aforementioned requirements for this.

Reply:
We discussed having a simple toggle at this week's meeting, since this feature for now is just implemented as an MVP. Will refine that piece further in #88. 

Good point about the length of time. But I'd disagree that this function is entirely redundant, but we may want to refine the mechanism by which we're checking for results, depending on the length of time and how we want to manage user concurrency.

Reply:
From Tanny: user ID - get request to the computing API - get all the jobs associated with the user ID and the segmentation - get all the jobs and the status of all these jobs - this happens on page load and happens at an interval.